desc/compute/_core.py

```diff
@@ -742,6 +658,54 @@ def compute_contravariant_metric_coefficients(
     return data
 
 
+def compute_toroidal_flux_gradient(
+    R_lmn,
+    Z_lmn,
+    Psi,
+    R_transform,
+    Z_transform,
+    data=None,
+    **kwargs,
+):
+    """Compute reciprocal metric coefficients.
+
+    Parameters
+    ----------
+    R_lmn : ndarray
+        Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate.
+    Z_lmn : ndarray
+        Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate.
+    Psi : float
+        Total toroidal magnetic flux within the last closed flux surface, in Webers.
+    R_transform : Transform
+        Transforms R_lmn coefficients to real space.
+    Z_transform : Transform
+        Transforms Z_lmn coefficients to real space.
+
+    Returns
+    -------
+    data : dict
+        Dictionary of ndarray, shape(num_nodes,) of toroidal flux gradient.
+
+    """
+    data = compute_toroidal_flux(Psi, R_transform.grid, data=data)
+    data = compute_contravariant_metric_coefficients(
+        R_lmn,
+        Z_lmn,
+        R_transform,
+        Z_transform,
+        data=data,
+    )
+
+    if check_derivs("grad(psi)", R_transform, Z_transform):
+        data["grad(psi)"] = data["psi_r"] * data["e^rho"].T
```

Comment:
I haven't thought hard about this, but could be missing a transpose. Can just close this if it's right.

Reply:
Good catch, thanks!